packages/devtools_app/lib/src/memory/flutter/memory_analyzer.dart

```diff
@@ -400,3 +587,84 @@ class _AnalysisFieldValueColumn extends ColumnData<AnalysisField> {
   @override
   double get fixedWidthPx => 250.0;
 }
+
+class ClassFields {
+  final List<String> _fields = [];
+
+  void clear() {
+    _fields.clear();
+  }
+
+  int get length => _fields.length;
+
+  void push(String name) {
+    _fields.add(name);
+  }
+
+  String pop() => _fields.removeLast();
+
+  String elementAt(int index) => _fields.elementAt(index);
+}
+
+const imageCache = 'ImageCache';
+
+/// Callback function when an ObjectReference's class name and fields all match.
+/// Parameters:
+///   className that matched
+///   fields that all matched
+///   value of the matched objectReference
+///
+typedef CompletedFunction = void Function(
+    String className, List<String> fields, dynamic value);
+
+class ObjectMatcher {
+  ObjectMatcher(this._matchCompleted);
+
+  static const Map<String, List<List<String>>> matcherDrillIn = {
+    '$imageCache': [
+      ['_pendingImages', 'data_', 'completer', 'context_', 'url'],
+      ['_cache', 'data_', 'completer', 'context_', 'url'],
+      ['_liveImages', 'data_', 'completer', 'context_', 'url'],
+    ]
+  };
+
+  final CompletedFunction _matchCompleted;
+
+  bool isClassMatched(String className) =>
+      matcherDrillIn.containsKey(className);
+
+  List<List<String>> _findClassMatch(String className) =>
+      matcherDrillIn[className];
+
+  /// First field name match.
+  bool findFieldMatch(ClassFields classFields, dynamic value) {
+    bool matched = false;
+
+    final className = classFields._fields.elementAt(0);
+    final listOfFieldsToMatch = _findClassMatch(className);
+
+    if (listOfFieldsToMatch != null) {
+      for (final fieldsToMatch in listOfFieldsToMatch) {
+        final fieldsSize = fieldsToMatch.length;
+        if (fieldsSize == classFields._fields.length - 1) {
```

Comment:
requiring all fields seems fragile. For example, we would miss a subclass that added a field and we would miss if an extra field was added to the class. Can we match if all fields are present or using a subclass relationship rather than requiring an exact match?

Reply:
I've added a todo it should be loosey goosey so we don't have to rev every time the framework or user's code changes (eventually they can define analysis too).

```
  // TODO(terry): Change to be less strict.  Look for subclass or parentage
  //              relationships.  If a new field or subclass is added we can
  //              still find what we're looking for.  Maybe even consider the
  //              the type we're looking for - best to be loosey goosey so
  //              we're not brittle as the Framework or any code changes.

```